pkg/webhook/util/configuration/configuration.go

```diff
@@ -58,65 +59,102 @@ func Ensure(kubeClient clientset.Interface, handlers map[string]types.HandlerGet
 		return err
 	}
 
-	var mutatingWHs []admissionregistrationv1.MutatingWebhook
-	for i := range mutatingTemplate {
-		wh := &mutatingTemplate[i]
-		wh.ClientConfig.CABundle = caBundle
-		path, err := getPath(&wh.ClientConfig)
-		if err != nil {
-			return err
-		}
-		if _, ok := handlers[path]; !ok {
-			klog.Warningf("Ignore webhook for %s in configuration", path)
-			continue
+	if utilfeature.DefaultFeatureGate.Enabled(features.EnableExternalCerts) {
+		// if using external certs, only check the caBundle of webhook
+		for i := range mutatingTemplate {
+			wh := &mutatingTemplate[i]
+			path, err := getPath(&wh.ClientConfig)
+			if err != nil {
+				return err
+			}
+			if _, ok := handlers[path]; !ok {
+				klog.Warningf("Ignore webhook for %s in configuration", path)
+				continue
+			}
+			if !bytes.Equal(wh.ClientConfig.CABundle, caBundle) {
+				return fmt.Errorf("caBundle of MutatingWebhookConfiguration %s does not match the external caBundle", mutatingWebhookConfigurationName)
+			}
 		}
-		if wh.ClientConfig.Service != nil {
-			wh.ClientConfig.Service.Namespace = webhookutil.GetNamespace()
-			wh.ClientConfig.Service.Name = webhookutil.GetServiceName()
 
-			if host := webhookutil.GetHost(); len(host) > 0 {
-				convertClientConfig(&wh.ClientConfig, host, webhookutil.GetPort())
+		for i := range validatingTemplate {
+			wh := &validatingTemplate[i]
+			path, err := getPath(&wh.ClientConfig)
+			if err != nil {
+				return err
+			}
+			if _, ok := handlers[path]; !ok {
+				klog.Warningf("Ignore webhook for %s in configuration", path)
+				continue
+			}
+			if !bytes.Equal(wh.ClientConfig.CABundle, caBundle) {
+				return fmt.Errorf("caBundle of ValidatingWebhookConfiguration %s does not match the external caBundle", mutatingWebhookConfigurationName)
 			}
```

Comment:
just return at the end of the if clause, and remove the else to make indentation smaller